spyder/plugins/editor/widgets/base.py

```diff
@@ -278,46 +278,111 @@ def unhighlight_current_cell(self):
         """Unhighlight current cell"""
         self.clear_extra_selections('current_cell')
 
-    #------Brace matching
-    def find_brace_match(self, position, brace, forward):
+    def in_comment(self, cursor=None, position=None):
+        """Returns true if the given position is inside a comment
+
+        Trivial default implementation. To be overridden by subclass.
+        This function is used to define the default behaciour of
+        self.find_brace_match.
+        """
+        return False
+
+    def in_string(self, cursor=None, position=None):
+        """Returns true if the given position is inside a string
+
+        Trivial default implementation. To be overridden by subclass.
+        This function is used to define the default behaciour of
+        self.find_brace_match.
+        """
+        return False
+
+    def find_brace_match(self, position, brace, forward,
+                         ignore_brace=None, stop=None):
+        """Returns position of matching brace
+
+        Parameters
+        ----------
+        position : int
+            The position of the brace to be matched.
+        brace : {'[', ']', '(', ')', '{', '}'}
+            The brace character to be matched.
+            [ <-> ], ( <-> ), { <-> }
+        forward : boolean
+            Whether to search forwards or backwards for a match.
+        ignore_brace : callable taking int returning boolean, optional
+            Whether to ignore a brace (as function of position).
+        stop : callable taking int returning boolean, optional
+            Whether to stop the search early (as function of position).
+
+        If both ignore_brace and brace are None, the default behaviour
+        then brace matching is handled differently depending on whether

```

Comment:
It seems there's a word missing in the sentence that starts after the comma, because I don't get its meaning.